Sources/ProjectDescription/TargetAction.swift

```diff
@@ -353,24 +454,56 @@ public struct TargetAction: Codable, Equatable {
                      basedOnDependencyAnalysis: basedOnDependencyAnalysis)
     }
 
+    /// Returns a target action that gets executed after the sources and resources build phase.
+    ///
+    /// - Parameters:
+    ///   - path: Path to the script to execute.
+    ///   - arguments: Arguments that to be passed.
+    ///   - name: Name of the build phase when the project gets generated.
+    ///   - inputPaths: List of input file paths.
+    ///   - inputFileListPaths: List of input filelist paths.
+    ///   - outputPaths: List of output file paths.
+    ///   - outputFileListPaths: List of output filelist paths.
+    ///   - basedOnDependencyAnalysis: Whether to skip running this script in incremental builds
+    /// - Returns: Target action.
+    public static func post(script: String,
+                            name: String,
+                            inputPaths: [Path] = [],
+                            inputFileListPaths: [Path] = [],
+                            outputPaths: [Path] = [],
+                            outputFileListPaths: [Path] = [],
+                            basedOnDependencyAnalysis: Bool? = nil) -> TargetAction
+    {
+        TargetAction(name: name,
+                     script: script,
+                     order: .post,
+                     inputPaths: inputPaths,
+                     inputFileListPaths: inputFileListPaths,
+                     outputPaths: outputPaths,
+                     outputFileListPaths: outputFileListPaths,
+                     basedOnDependencyAnalysis: basedOnDependencyAnalysis)
+    }
+
     // MARK: - Codable
 
     public init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
         name = try container.decode(String.self, forKey: .name)
         order = try container.decode(Order.self, forKey: .order)
-        arguments = try container.decode([String].self, forKey: .arguments)
         inputPaths = try container.decodeIfPresent([Path].self, forKey: .inputPaths) ?? []
         inputFileListPaths = try container.decodeIfPresent([Path].self, forKey: .inputFileListPaths) ?? []
         outputPaths = try container.decodeIfPresent([Path].self, forKey: .outputPaths) ?? []
         outputFileListPaths = try container.decodeIfPresent([Path].self, forKey: .outputFileListPaths) ?? []
         basedOnDependencyAnalysis = try container.decodeIfPresent(Bool.self, forKey: .basedOnDependencyAnalysis)
-        if let path = try container.decodeIfPresent(Path.self, forKey: .path) {
-            self.path = path
-            tool = nil
+
+        let arguments: [String] = try container.decodeIfPresent([String].self, forKey: .arguments) ?? []
```

Comment:
I'd add tests with `XCTAssertCodable` to test the new logic

Reply:
I added a test for that [here](https://github.com/tuist/tuist/pull/2192/files#diff-7da80918f809619ac5ddf18f54c4a8f9057a5223c41d5d18781baa5cdc56e232R37). Is there additional logic I should test?